platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/lazy_configuration.adoc

```diff
@@ -321,6 +321,30 @@ $ gradle show
 include::{snippetsPath}/providers/propertyConvention/tests/propertyConvention.out[]
 ----
 
+=== Where to apply conventions from?
+
+In general, you want to apply convention to a property at configuration time, before tasks execute, so task actions are not appropriate.
```

Comment:
🤔 I think this deserves some justification. When applied in action:
* the convention value is not going to participate in input snapshotting.
* it may override another convention specified externally.

Reply:
Thanks, Mike,. I think this (what to do at configuration time vs execution) should be covered elsewhere, not here. I think I will just refrain from mentioning what not to do, and cover the cases we want to promote.